node/GUIDE.md

```diff
@@ -0,0 +1,230 @@
+endpoints on your tchannel.
+
+First we create a sub channel using [`channel.makeSubChannel()`](./docs/sub-channels.md#channelmakesubchanneloptions).
+
+Once we have a sub channel we have to create a [`TChannelThrift(opts)`](./docs/as-thrift.md#var-tchannelthrift--tchannelthriftops) codec.
+
+Finally you can call [`.register()`](./docs/as-thrift.md#tchannelthriftregistertchannel-arg1-ctx-handlerfn) on the thrift
+codec to register your actual endpoints.
+
+```js
+var TChannelThrift = rootChannel.TChannelAsThrift;
+
+var keyChan = rootChannel.makeSubChannel({
+    serviceName: 'keyvalue'
+});
+var keyThrift = TChannelThrift({
+    source: fs.readFileSync('./keyvalue.thrift', 'utf8')
+});
+var ctx = {
+    store: {}
+};
+
+keyThrift.register(keyChan, 'KeyValue::get_v1', ctx, get);
+keyThrift.register(keyChane, 'KeyValue::put_v1', ctx, put);
+
+function get(context, req, head, body, cb) {
+    cb(null, {
+        ok: true,
+        body: {
+            value: context.store[body.key]
+        }
+    });
+}
+function put(context, req, head, body, cb) {
+    context.store[body.key] = body.value;
+    cb(null, {
+        ok: true,
+        body: null
+    });
+}
+```
+
+### Registering with hyperbahn
+
+Create a Hyperbahn client using [`HyperbahnClient(opts)`](./docs/hyperbahn.md#var-hyperbahnclient--hyperbahnclientoptions).
+
+Call [`.advertise()`](./docs/hyperbahn.md#hyperbahnclientadvertise)
+on the hyperbahn client to advertise your service with hyperbahn.
+
+```js
+var HyperbahnClient = require('tchannel/hyperbahn/');
+
+var hyperbahnClient = HyperbahnClient({
+    tchannel: rootChannel,
+    serviceName: 'keyvalue',
+    hostPortList: ['127.0.0.1:21301'],
+    hardFail: true
+});
+
+hyperbahnClient.advertise();
+hyperbahnClient.once('advertised', onAdvertised);
+
+function onAdvertised() {
+    /* hooray! */
+}
+```
+
+### Local testing
+
+Your service is now available on hyperbahn. You can test this by
+making a call using [tcurl](https://github.com/uber/tcurl)
+
+```sh
+node tcurl.js -p [HYPERBAHN-HOSTPORT] -t [DIR-TO-THRIFT] keyvalue KeyValue::put_v1 -3 '{"key":"hello","value":"world"}'
+node tcurl.js -p [HYPERBAHN-HOSTPORT] -t [DIR-TO-THRIFT] keyvalue KeyValue::get_v1 -3 '{"key":"hello"}'
+```
+
+Make sure to replace `[HYPERBAHN-HOSTPORT]` with one of the
+hyperbahn instance host ports and `[DIR-TO-THRIFT]` with a folder
+where the thrift files are stored.
+
+Your service can be accessed over Hyperbahn + TChannel from any
+language.
+
+## Writing a tchannel client
+
+To make a client that talks to hyperbhan, you need to:
+
+ - Create a tchannel
+ - Create a hyperbahn client
+ - Making outgoing call requests using TChannelThrift codec
+
+### Creating a tchannel
+
+TChannel is a bidirectional RPC library; you already know how
+to create a root channel.
+
+A pure client does not need to call `.listen()` but it's
+recommended that you `.listen()` anyway and `advertise()` anyway.
+It's always best to get onto the hyperbahn as early as possible.
```

Comment:
The normal case is for you to listen and use incoming connections. If you do not do this you will use random outgoing connections.

There is no reason for pure clients to advertise; but there should be no pure clients.